compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -2372,6 +2375,60 @@ impl<'a> PartialEvaluator<'a> {
         result_value
     }
 
+    fn reset_qubits(
```

Comment:
I wonder if it's better to have this logic here or to instead update `eval_expr_call_to_intrinsic_qis` to take a parameter for the callable type and then just call through to that. On the one hand, this gives us a place to put other special handling if need it. On the other hand, this is basically the same as `eval_expr_call_to_intrinsic_qis` with the only difference seeming to be callable type. Unless I'm missing some other difference?

Reply:
I will have a better answer for this once Measurements and Resets are unified. I would like to keep the logic that handles attributes together. Let's wait until the unification PR to decide on this.